### PR TITLE
Functionality to AddTask and pass data to TaskViewController

### PR DESCRIPTION
**Pull request by jWarrenDev**
# Description
Added Features to AddTaskVC. Ability to Search Categories fetched from server, to add a corresponding task, and to pass that Task to the Task View Controller through a Storyboard Reference. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status
- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested by Myself, waiting for review 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+class AddTaskViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UITextFieldDelegate {
+   
+
+    // Load sample values
+    
+    
+    
+    var sampleCategories = ["Kitchen", "Bathroom", "Dining Room", "Living Room" ,"Master Bedroom"]
+    var sampleTask = ["Clean up", "Wipe Down", "Wash dishes", "Vaccum", ]
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: Remove test code

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            
+           
+            print(task)
+            print(category)
+            
+        
+            destinationVC.taskController = taskController
+            destinationVC.task = task
+            destinationVC.category = category
+            //destinationVC.household = household
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: Need to be uncommented

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    
+    var sampleCategories = ["Kitchen", "Bathroom", "Dining Room", "Living Room" ,"Master Bedroom"]
+    var sampleTask = ["Clean up", "Wipe Down", "Wash dishes", "Vaccum", ]
+    
+    // Properties
+    var categories: [Category]?
+    var userController: UserController?
+    var currentUser: User?
+    var householdController: HouseholdController?
+    var taskController = TaskController()
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: shouldn't be initialized... initialized in TabBarVC

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    var categoryController = CategoryController()
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: shouldn't be initialized... initialized in TabBarVC

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    
+    var task : Task?
+    var catID : UUID?
+    var category : Category?
+    var household : Household?
+    
+    //∫???
+    // let tabBarController = TabViewViewController.self
+    
+    let mainCategoriesViewController = MainCategoriesViewController()
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift:26]: Why is the MainCategoriesVC being initialized here? Possible left over from earlier code plans?

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift (hunk 4 above)
Comment (nateyoungren) [reply ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: Category isn't being passed from TabBarVC right now, but we need to refactor that. This is probably why it didn't work without initialization. @jWarrenDev

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        
+        categoryTableView.rowHeight = UITableView.automaticDimension
+        
+        if let tabBar = self.tabBarController as? TabViewViewController {
+        
+            self.taskController = tabBar.taskController
+            self.currentUser = tabBar.currentUser
+            self.householdController = tabBar.householdController
+            self.userController = tabBar.userController
+
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift:62]: We will set categoryController from the tabBar on this line. @jWarrenDev

**Review by nateyoungren — CHANGES_REQUESTED** [review]
Make the changes as if the TabBarVC does have the CategoryController... Ignore the errors that it will produce. We will initialize it in the TabBarVC later.

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift (hunk 6 above)
Comment (jWarrenDev) [reply ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift:62]: ok Cool

**Review by nateyoungren — APPROVED** [review]

File: ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift [hunk 7]
```diff
@@ -10,27 +10,19 @@ import UIKit
 
 class AddTaskViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UITextFieldDelegate {
    
-
```
Comment (nateyoungren) [inline ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift]: Pull and merge first

**Review by nateyoungren — CHANGES_REQUESTED** [review]
Please pull first

**Review by nateyoungren — APPROVED** [review]

**Pull request merged**